### DISCUSSION TITLE
MM2 offset translation from Confluent Kafka is not working.

### DISCUSSION THREAD

**Discussion by AAR256** [strimzi/strimzi-kafka-operator#9371] [Q&A]
Hi,

To provide the context, we have few Strimzi Kafka clusters running in Production with MirrorMaker 2 in place to enable HA. We also have a confluent kafka cluster in which some of the topics are replicated to Strimzi cluster through similar MM2 config. The issue is that offset translation for consumer groups is not happening for the replicated topics from Confluent cluster which isn't the case with the others. 

<img width="337" alt="MicrosoftTeams-image (8)" src="https://github.com/strimzi/strimzi-kafka-operator/assets/95288537/19e83d8a-ab19-46f9-9d5c-0ccf9cc2f3ae">

The checkpoint topic doesn't have any events as well. 

There is no problem in replication though. The events are replicating just fine. Also to note, we're using the same MM2 configuration across all MM2 clusters which is slightly tuned boiler plate code available in Strimzi repo. 
If anything else is required to debug this, please feel free to let me know.

**Comment by scholzj** [comment]
Strimzi just repackages Apache Kafka and its MM2. It does not add any custom features for syncing the offsets. I think Confluent Platform / Confluent Cloud does not use open-source Apache Kafka. As far as I know, they have their own fork which might have some differences. So maybe this is one of them? If they for example store the consumer offsets differently, they might not be visible to MM2. It is not clear what kind of Confluent Kafka you use. I think some incompatibilities would be more likely for Confluent Cloud.

**Reply by AAR256** [reply]
@scholzj That makes a lot of sense. I went through the configuration across all the setups. The speculated reason behind this seems to be due to the missing ```__consumer-offsets``` topic which has been hidden in Confluent Cloud. Due to security concerns they're not willing to expose the topic publicly within the cluster. I really don't see any other option here though.

**Reply by scholzj** [reply]
Yeah, if you don't see it, then the offset sync will definitely not work as that is what it is using.
